File: query.js
'use strict';

var util = require('util');
var waterfall = require('async-waterfall');
var hana = require('@sap/hana-client');

var connOptions = {
    serverNode: '<endpoint>',
	encrypt: 'true',
    uid: '<user>',
    pwd: '<password>'
};

var connection = hana.createConnection();

var tasks = [
	myconn,
    mysql, myexecute, myresults,
    mydisco
	];

waterfall(tasks, done);
console.log("Async calls underway\n");

function myconn(cb) {
    connection.connect(connOptions);
    cb(null);
}

function mysql(cb) {
	var sql = 'SELECT * FROM travel.budgetrooms';
    cb(null, sql);
}

function myexecute(sql, cb) {
    var rows = connection.exec(sql);
    cb(null, rows);
}

function myresults(rows, cb) {
    console.log(util.inspect(rows, { colors: true }));
    console.log("");
    cb(null);
}

function mydisco(cb) {
    connection.disconnect(cb);
}

function done(err) {
    console.log("Async done");
    if (err) {
        return console.error(err);
    }
}

File: query.java
import java.sql.*;
public class query {
    public static void main(String[] argv) {
        Connection connection = null;
        try {
            connection = DriverManager.getConnection("jdbc:sap://<endpoint>/?encrypt=true", "<user>", "<password>");
        }
        catch (SQLException e) {
            System.err.println("Connection to SAP HANA Cloud Failed:");
            System.err.println(e);
            return;
        }
        if (connection != null) {
            try {
                System.out.println("Connected to SAP HANA Cloud!");
                Statement stmt = connection.createStatement();
                ResultSet resultSet = stmt.executeQuery("SELECT * FROM travel.budgetrooms");
                while (resultSet.next()) {
                    System.out.println(resultSet.getString(1) + " " + resultSet.getString(2) + " " + resultSet.getString(3) + " " + resultSet.getString(4) + " " + resultSet.getString(5) + " " + resultSet.getString(6));
                }
            }
            catch (SQLException e) {
                System.err.println("Query failed!");
            }
        }
    }
}
